cmd/worker/main.go

```diff
@@ -153,34 +152,30 @@ func handleMessage(ctx context.Context, msg *pubsub.Message, packagesBucket *blo
 		return err
 	}
 
+	staticSandboxOpts := append(worker.StaticSandboxOptions(), sandboxOpts...)
 	dynamicSandboxOpts := append(worker.DynamicSandboxOptions(), sandboxOpts...)
-	result, err := worker.RunDynamicAnalysis(ctx, pkg, dynamicSandboxOpts, "")
-	if err != nil {
-		return err
-	}
 
-	staticSandboxOpts := append(worker.StaticSandboxOptions(), sandboxOpts...)
-	var staticResults analysisrun.StaticAnalysisResults
-	// TODO run static analysis first and remove the if statement below
-	if resultStores.StaticAnalysis != nil {
-		staticResults, _, err = worker.RunStaticAnalysis(ctx, pkg, staticSandboxOpts, staticanalysis.All)
-		if err != nil {
-			return err
-		}
+	// run both dynamic and static analysis regardless of error status of either
+	// and return combined error(s) afterwards, if applicable
+	staticResults, _, staticAnalysisErr := worker.RunStaticAnalysis(ctx, pkg, staticSandboxOpts, staticanalysis.All)
+	if staticAnalysisErr == nil {
+		staticAnalysisErr = worker.SaveStaticAnalysisData(ctx, pkg, resultStores, staticResults)
```

Comment:
Is there a rationale for changing the order of static analysis and dynamic analysis?

Reply:
Originally I only put static after dynamic because it was conditionally run, I think it makes sense to have static analysis running first because

1. Static analysis requires downloading the package archive and this may end up being able to be reused for dynamic analysis. The saving analyzed package feature can also potentially reuse this download 
2. For some reason, static analysis data was saved before dynamic analysis before, so now that the saving is paired with the running, it means that the order of saving results is the same. 
3. static analysis is usually faster than dynamic analysis 

All that said, it shouldn't make a huge difference. If you feel strongly that the order should be kept as is, we can do that.

Reply:
All good. I appreciate understanding the rationale and it helps to have context in the PR for the future.

Reply:
No worries and I agree it's good to have the context here too :)